parlai/mturk/core/react_server/dev/components/core_components.jsx

```diff
@@ -768,37 +775,145 @@ class TaskDescription extends React.Component {
   }
 }
 
+class ContextView extends React.Component {
+  render () {
+    // TODO pull context title from templating variable
+    let header_text = 'Context';
+    let context = (
+      'To render context here, write or select a ContextView ' +
+      'that can render your task_data, or write the desired ' +
+      'content into the task_data.html field of your act');
+    if (this.props.task_data !== undefined &&
+        this.props.task_data.html !== undefined) {
+      context = this.props.task_data.html;
+    }
+    return (
+      <div>
+          <h1>{header_text}</h1>
+          <hr style={{'borderTop': '1px solid #555'}} />
+          <span
+            id="context" style={{'fontSize': '16px'}}
+            dangerouslySetInnerHTML={{__html: context}}
+          />
+      </div>
+    );
+  }
+}
+
 class LeftPane extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {current_pane: 'instruction', last_update: 0};
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState){
+    if (nextProps.task_data.last_update !== undefined &&
+        nextProps.task_data.last_update > prevState.last_update) {
+      return {current_pane: 'context',
+              last_update: nextProps.task_data.last_update};
+    }
+    else return null;
+  }
+
   render () {
     let v_id = this.props.v_id;
     let frame_height = this.props.frame_height;
     let frame_style = {
       height: frame_height + 'px',
       'backgroundColor': '#dff0d8',
       padding: '30px',
-      overflow: 'auto'
+      overflow: 'auto',
     };
     let XTaskDescription = getCorrectComponent('XTaskDescription', v_id);
     let pane_size = this.props.is_cover_page ? 'col-xs-12' : 'col-xs-4';
-    return (
-      <div id="left-pane" className={pane_size} style={frame_style}>
+    let has_context = this.props.task_data.last_update !== undefined;
+    if (this.props.is_cover_page || !has_context) {
+      return (
+        <div id="left-pane" className={pane_size} style={frame_style}>
+            <XTaskDescription {...this.props} />
+        </div>
+      );
+    } else {
+      let XContextView = getCorrectComponent('XContextView', v_id);
+      // In a 2 panel layout, we need to tabulate the left pane to be able
+      // to display both context and instructions
+      let nav_items = [
+        <NavItem
+          eventKey={'instruction'}
+          key={'instruction-selector'}
+          title={'Task Instructions'}>
+          {'Task Instructions'}
+        </NavItem>,
+        <NavItem
+          eventKey={'context'}
+          key={'context-selector'}
+          title={'Context'}>
+          {'Context'}
+        </NavItem>
+      ];
+      let display_instruction = {
+        'backgroundColor': '#dff0d8',
```

Comment:
might be good to standardize this color somewhere since it appears a few times in these files

Reply:
Yup - I'm planning to migrate this to a css file once everything is properly supported. Should make things better to have style separate from everything else.